source/installation/resource-manager/kubernetes.rst

```diff
@@ -0,0 +1,189 @@
+namespace_prefix
+  The prefix to your namespace. Use this prefix if you have assertions on what namespaces
+  are available. I.e., a Kyverno policy that ensures all namespaces are ``user-\w+``.
+all_namespaces
+  A boolean to determine if the user will query for pods in other namespaces.  When false
+  users will only query in their namespace. If true they will query and display pods from
+  all namespaces.
+auto_supplemental_groups:
+  Automatically populate a container's ``securityContext.supplementalGroups`` with the users
+  supplemental groups.
+server
+  The kubernetes server to communicate with.  This field is a map with ``endpoint`` and
+  ``cert_authority_file`` keys.
+auth:
+  See the notes on `Authentication`_ below.
+mounts:
+  Site wide mount points for all kubernetes jobs. See the 
+  :ref:`documentation on kubernetes mounts <kubernetes-mounts>` for more details.
+
+Per User Kubernetes
+*******************
+
+To get kubernetes to act like a Per User resource there are some conventions
+we put in place. Users only schedule pods in their own namespaces
+and they always run those pods as themselves.
+
+At most users could be allow to read pods from other namespaces (have sufficient
+privileges to run ``kubectl get pods --all-namespaces``), but this is not required.
+Being able to view pods in other namespaces is only applicable to a feature like
+viewing active jobs and seeing pods from other namespaces in that view.
+
+Second is that we specify the `Kubernetes security context`_ such that pods have
+the same UID and GID as the actual user.
+
+Open OnDemand will always use the users UID and GID as the runAsUser and runAsGroup.
+fsGroup is always the same as runAsGroup. runAsNonRoot is always set to true.
+supplementalGroups are empty by default. One can automatically populate them with a
+cluster configuration above or specify them for each app individually.
+
+You should have policies in place to enforce these.
+
+Bootstrapping an Open OnDemand Kuberenetes cluster
+**************************************************
+
+We ship with `open ondemand provided hooks`_ to bootstrap users when the login
+to Open OnDemand. These scripts will create their namespace, a networking policy,
+and rolebindings for user and the service accounts in their namespace.
+
+The namespace is just to create their namespace (perhaps with a prefix configured
+above). A user ``oakley`` would create the ``oakley`` namespace. If you've configured
+to use prefix ``user-``, then the namespace would be ``user-oakley``.
+
+The networking policy ensures that pods cannot communicate inbetween namespaces.
+
+The rolebindings bind the user, ``oakley`` in this case, to the ``ood-user``
+ClusterRole.  It also creates a service account in the namespace ``user-oakley``.
+The service account is used in initializing configmaps and secrets for the Open
+OnDemand pods.
+
+**TODO** Need to find, share and document the ood-user ClusteRole.
```

Comment:
https://github.com/OSC/ondemand/blob/master/hooks/k8s-bootstrap/ondemand.yaml

Reply:
I think I fixed this, reworded it a bit to fit around the link.